pkg/scheduler/priorities.go

```diff
@@ -132,3 +134,92 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod api.Pod, podLister
+// two percentage metrics are to each other.
+// Details: score = 1 / sqrt[(cpu_pct - average)^2 + (mem_pct - average)^2], average = (cpu_pct + mem_pct) / 2
+func BalancedResourceAllocation(pod api.Pod, podLister PodLister, minionLister MinionLister) (HostPriorityList, error) {
+	nodes, err := minionLister.List()
+	if err != nil {
+		return HostPriorityList{}, err
+	}
+	podsToMachines, err := MapPodsToMachines(podLister)
+
+	list := HostPriorityList{}
+	for _, node := range nodes.Items {
+		list = append(list, calculateBalancedResourceAllocation(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func calculateBalancedResourceAllocation(pod api.Pod, node api.Node, pods []api.Pod) HostPriority {
+	totalMilliCPU := int64(0)
+	totalMemory := int64(0)
+	score := int(0)
+	for _, existingPod := range pods {
+		for _, container := range existingPod.Spec.Containers {
+			totalMilliCPU += container.Resources.Limits.Cpu().MilliValue()
+			totalMemory += container.Resources.Limits.Memory().Value()
+		}
+	}
+	// Add the resources requested by the current pod being scheduled.
+	// This also helps differentiate between differently sized, but empty, minions.
+	for _, container := range pod.Spec.Containers {
+		totalMilliCPU += container.Resources.Limits.Cpu().MilliValue()
+		totalMemory += container.Resources.Limits.Memory().Value()
+	}
+
+	capacityMilliCPU := node.Status.Capacity.Cpu().MilliValue()
+	capacityMemory := node.Status.Capacity.Memory().Value()
+
+	cpuPctOfCapRequested := calculatePctOfCapRequested(totalMilliCPU, capacityMilliCPU, node.Name)
+	memoryPctOfCapRequested := calculatePctOfCapRequested(totalMemory, capacityMemory, node.Name)
+	avergePctOfCapRequested := (cpuPctOfCapRequested + memoryPctOfCapRequested) / 2
+	cpuDiff := cpuPctOfCapRequested - avergePctOfCapRequested
+	memoryDiff := memoryPctOfCapRequested - avergePctOfCapRequested
+	diff := math.Sqrt(math.Pow(cpuDiff, 2) + math.Pow(memoryDiff, 2))
+	// closer percentage metrics lead to higher score with 10 being the highest
+	if diff <= 0.1 {
+		score = 10
+	} else {
+		score = int(1 / diff)
+	}
+	// if Requesetd/Capacity exceeds 100%, the PctOfCapRequested is set to 1 (100%) and the host should never be preferred.
+	// isEqual() is used to eliminate the risk of value inaccuracy.
+	// Higher threhold of the acccuracy parameter is beyond the scope of BRU algorithm; as said in the introduction
+	// this BRU PriorityFunction should NEVER be used alone.
+	if isEqual(cpuPctOfCapRequested, float64(1)) || isEqual(memoryPctOfCapRequested, float64(1)) {
+		score = 0
+	}
+
+	glog.V(4).Infof(
+		"%v -> %v: Balanced Resource Allocation, Absolute/Requested: (%d, %d) / (%d, %d) Score: (%d)",
+		pod.Name, node.Name,
+		totalMilliCPU, totalMemory,
+		capacityMilliCPU, capacityMemory,
+		score,
+	)
+
+	return HostPriority{
+		host:  node.Name,
+		score: score,
+	}
+}
+
+func calculatePctOfCapRequested(requested, capacity int64, node string) float64 {
```

Comment:
How about fractionOfCapacity for a name, or capacityFraction?

(Reasoning: all functions "calculate" things, so there's no need to state so in the name, and fraction is clearer than percent, which could leave the reader thinking you've multiplied the result by 100.)